core/src/banking_stage/committer.rs

```diff
@@ -133,29 +134,45 @@ impl Committer {
         starting_transaction_index: Option<usize>,
     ) {
         if let Some(transaction_status_sender) = &self.transaction_status_sender {
+            let sanitized_transactions = batch.sanitized_transactions();
+
             // Clone `SanitizedTransaction` out of `RuntimeTransaction`, this is
             // done to send over the status sender.
-            let txs = batch
-                .sanitized_transactions()
+            let txs = sanitized_transactions
                 .iter()
                 .map(|tx| tx.as_sanitized_transaction().into_owned())
                 .collect_vec();
+
             let post_balances = bank.collect_balances(batch);
             let post_token_balances =
                 collect_token_balances(bank, batch, &mut pre_balance_info.mint_decimals);
+
             let mut transaction_index = starting_transaction_index.unwrap_or_default();
-            let batch_transaction_indexes: Vec<_> = commit_results
+            let (batch_transaction_indexes, tx_costs): (Vec<_>, Vec<_>) = commit_results
                 .iter()
-                .map(|commit_result| {
-                    if commit_result.was_committed() {
+                .zip(sanitized_transactions.iter())
+                .map(|(commit_result, tx)| {
+                    if let Ok(committed_tx) = commit_result {
                         let this_transaction_index = transaction_index;
                         saturating_add_assign!(transaction_index, 1);
-                        this_transaction_index
+
+                        let tx_cost = Some(
+                            CostModel::calculate_cost_for_executed_transaction(
```

Comment:
note to self. RPC-only performance impact. Not sure how expensive this calculation is.